### DISCUSSION TITLE
How about an image manager for the web interface?

### DISCUSSION THREAD

**Discussion by nzyeehaa** [datenstrom/community#716] [Ask a question] [answered]
drag and drop image upload works great.
local computer select image upload works great.

but the one thing missing is a way to manage the images on the server - from the wysiwyg editor....

(EDIT : caution: with the following link, I am only referring to the **image management functions** - not the rest of the fluff, yellows editor is fine (@schulle4u ) )

[check out this](https://xdsoft.net/jodit/) as a possible addition/implementation to manage the images/files already on the server, 

- drag and drop +
- local computer select +
- **server image select +**
- **server folder management +**
- **server image management +**
- **server image resizing +**
- **server image cropping**

**Comment by IngoBelka** [comment]
[#440 (comment)](https://github.com/datenstrom/yellow/discussions/440#discussioncomment-1917681)

Greetings Ingo

**Reply by nzyeehaa** [reply]
Hi Ingo,
For the whole site, and code adjustments, layouts, css, I use my server's filemanager, or sometimes compress the site, download and edit locally, then upload again.

But for the website **users**, it needs to be really simple, they use the web browser, and I add an image/file manager... (usually [TFM](https://github.com/prasathmani/tinyfilemanager), [SFM](https://github.com/jcampbell1/simple-file-manager), responsivefilemanager (which may not be a good choice currently),
which allows them to...
- reuse images
- delete images
- organise images
- rename images
- automatically resize on upload
- set different sizes per folder (i.e: slides are 1024px, images are 640px)

but this doesnt advance yellow and make it better/simpler. 
The current Wysiwg/editor is brilliant, but someway of the **user being able to manage their images** needs to be integrated, and the example i gave above does it brilliantly.
put the jodit editor aside, and look at the **image management** side...
- upload OR browse
- edit
- resize
- crop
- duplicate
- rename
- delete
- folders
- preview
- download
if I get time I will have a crack at it myself, as it will be better than adding an external filemanager (with added security risk)

**Comment by markseuffert** [comment, marked as answer]
Here's what we discussed on Discord. I understand that some users want to do more in the web browser. I also have a couple wishes for the editor and hopefully will add some features to the next release. If anyone is worried about feature creep, that image management would add too much functionality and that not all users need this kind of extra functionality. You are right. For now the [edit extension](https://github.com/annaesvensson/yellow-edit) does everything. It provides all the editing functionality for the web browser. But if the number of features become too large for one extension, we can break it down into individual extensions. So you can add only the features you need. 

I think this would fit the spirit of Datenstrom Yellow.

**Comment by annaesvensson** [comment]
At the moment no developer is working on this, but would be lovely to have something like this in the future. Maybe someone wants to [make an extension](https://github.com/annaesvensson/yellow-publish) or improve the [edit extension](https://github.com/annaesvensson/yellow-edit) with image management features? All that is needed is someone to volunteer. 😄
